src/Datasource/Type.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.2.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Datasource;
+
+class Type
+{
+
+    /**
+     * List of basic type mappings, used to avoid having to instantiate a class
+     * for doing conversion on these
+     *
+     * @var array
+     * @deprecated 3.1 All types will now use a specific class
+     */
+    protected static $_basicTypes = [
+        'string' => ['callback' => ['\Cake\Datasource\Type', 'strval']],
+        'text' => ['callback' => ['\Cake\Datasource\Type', 'strval']],
+        'boolean' => [
+            'callback' => ['\Cake\Datasource\Type', 'boolval']
```

Comment:
I left this is because @lorenzo left it in as well. I guess there's still a use for this. https://github.com/cakephp/cakephp/blob/7c3647c2776431b2fce8e0fa3a0e184c0a13a0ec/src/Database/Type.php#L57